pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -95,8 +123,20 @@ func NewMCDefaultRouteController(
 		ciImportListerSynced:         ciImportInformer.Informer().HasSynced,
 		queue:                        workqueue.NewNamedRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(minRetryDelay, maxRetryDelay), "gatewayroute"),
 		installedCIImports:           make(map[string]*mcv1alpha1.ClusterInfoImport),
-		enableStretchedNetworkPolicy: enableStretchedNetworkPolicy,
-		enablePodToPodConnectivity:   enablePodToPodConnectivity,
+		namespace:                    multiclusterConfig.Namespace,
+		enableStretchedNetworkPolicy: multiclusterConfig.EnableStretchedNetworkPolicy,
+		enablePodToPodConnectivity:   multiclusterConfig.EnablePodToPodConnectivity,
+		serviceCIDRProvider:          serviceCIDRProvider,
+	}
+	if multiclusterConfig.TrafficEncryptionMode == TrafficEncryptionModeWireGuard {
+		if multiclusterConfig.WireGuard.Port == 0 {
+			multiclusterConfig.WireGuard.Port = multiclusterWireGuardPort
+
+		}
```

Comment:
I mean you can add this check to the `validateMulticlusterConfig` func in the cmd/antrea-agent/options.go
I feel you can add the whole `WireGuardConfig` initialization in the options.go and pass it to the controller?

Reply:
I'm not sure if we can do initialization in `cmd/antrea-agent/options.go`, since we just do validation in that file. Maybe we could add a complete function?